test/e2e/utils/pod.go

```diff
@@ -17,37 +17,84 @@ limitations under the License.
 package utils
 
 import (
-         "sync"
+	"sync"
 
-         "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo"
 
-	 v1 "k8s.io/api/core/v1"
-         metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-         "k8s.io/kubernetes/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
 )
 
 const (
 	CentosImage = "quay.io/centos/centos:8"
 )
 
+func MakeGuaranteedSleeperPod(cpuLimit string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "sleeper-gu-pod",
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{
+				v1.Container{
+					Name:  "sleeper-gu-cnt",
+					Image: CentosImage,
+					// 1 hour (or >= 1h in general) is "forever" for our purposes
+					Command: []string{"/bin/sleep", "1h"},
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							// we use 1 core because that's the minimal meaningful quantity
```

Comment:
How is this is a guaranteed pod if there is no resource request. For a pod to be guaranteed we have to make sure request=limit ?

Reply:
because https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits - see note at the bottom of the paragraph.

Reply:
Ah nice! Today I learned :) Thanks